packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -239,6 +239,7 @@ export class TeamsBotV2Impl {
       try {
         await utils.execute("npm install", workingPath);
         await utils.execute("npm run build", workingPath);
+        await utils.execute("npm prune --production", workingPath);
```

Comment:
NIT: I'm wondering if it will make any impact on the local debugging lifecycle. Consider the case when customers run local debug after deployment.

Reply:
npm prune --production will clear all dev dependencies, when project lang is TS, that means the typescript package will be removed in this process.
I don't know if npm install will run before the local debug?

Reply:
What about `npm install --production` and then `npm run build`?

Reply:
I prefer not to remove anything from user's project. 
Maybe we can prune in CD environment?

Reply:
Do you mean we do "npm prune --production" after the local build method but before we package the zip that we want to upload?
Or we can just run npm install again after the upload process?

Reply:
From my perspective, the goal you want to achieve is packing the project for remote in a production/release build style which may do extra optimizations. But the local debug doesn't need that. To achieve both local and remote goal, try thinking about Asp.Net project's build types of Debug and Release.

For a long term consideration, TeamsFx project may need some kind of Build Type abstractions. 
For now, I prefer not to do this kind of optimizations to avoid affecting the local debug experience.